### PR TITLE
samples: nrf9160: gnss: add reference position support

### PR DESCRIPTION
**Pull request by tokangas**
Added support for reference position. When the reference position is configured, the sample calculates the distance from the reference position for each fix.

### CODE REVIEW

**Review by peknis — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]

File: samples/nrf9160/gnss/src/main.c [hunk 1]
```diff
@@ -241,6 +278,11 @@ static void ttff_test_got_fix_work_fn(struct k_work *item)
 	if (time_blocked > 0) {
 		LOG_INF("Time GNSS was blocked by LTE: %u", time_blocked);
 	}
+	if (ref_used) {
+		LOG_INF("Distance from reference: %.01f",
```
Comment (hiltunent) [inline samples/nrf9160/gnss/src/main.c:282]: Or you could call print_distance_from_reference() here?

**Comment by hiltunent** [comment]
Damn, I should have finished my review before lunch. Now it's merged!

File: samples/nrf9160/gnss/src/main.c (hunk 1 above)
Comment (tokangas) [reply samples/nrf9160/gnss/src/main.c:282]: The reason why I didn't call that was that the function used `printk()`, while here I wanted to use logger. Now that I wrote this comment, I realized that I could have just selected between `printk()` and `LOG_INF()` in `print_distance_from_reference()` based on the configuration.

File: samples/nrf9160/gnss/src/main.c [hunk 2]
```diff
@@ -64,11 +73,39 @@ BUILD_ASSERT(IS_ENABLED(CONFIG_LTE_NETWORK_MODE_LTE_M_GPS) ||
...
+	double lat1_rad = lat1 * PI / 180.0;
+	double lat2_rad = lat2 * PI / 180.0;
+
+	double a = pow(sin(d_lat_rad / 2), 2) +
+		   pow(sin(d_lon_rad / 2), 2) *
+		   cos(lat1_rad) * cos(lat2_rad);
+
+	double c = 2 * asin(sqrt(a));
+
+	return 6371.0 * 1000.0 * c;
```
[the remark this answers is not in the document]
Comment (tokangas) [reply samples/nrf9160/gnss/src/main.c:106]: Yes, that is true, a define would be nicer.

**Pull request merged**